pcdsdevices/smarpod.py

```diff
@@ -0,0 +1,266 @@
+"""
+Module for the SmarPod and related devices.
+"""
+from ophyd.device import Component as Cpt
+from ophyd.device import Device
+from ophyd.signal import EpicsSignal
+from .interface import BaseInterface
+
+
+
+class SmarPod(BaseInterface, Device):
+    """
+    A base class for a SmarPod controller.
+    """
+    init = Cpt(EpicsSignal, ':CMD:INIT', kind='normal')
+    status = Cpt(EpicsSignal, ':CMD:ALL_STATUS', kind='normal')
+
+
+
+    """
+    Controlling unit number
+    """
+    cmd_unit = Cpt(EpicsSignal, ':CMD:UNIT', kind='normal')
+    unit = Cpt(EpicsSignal, ':UNIT', kind='normal', doc='Selected SmarPod unit')
+
+
+
+    """
+    Reference
+    """
+    cmd_ref = Cpt(EpicsSignal, ':CMD:REF', kind='normal')
+    ref = Cpt(EpicsSignal, ':REF', kind='normal', doc='Readback referenced state')
+    _ref = Cpt(EpicsSignal, ':_REF', kind='normal', doc='Async reference response') 
+    cmd_ref_method = Cpt(EpicsSignal, ':CMD:REF_METHOD', kind='normal', doc='Set reference method') 
+    ref_method = Cpt(EpicsSignal, ':REF_METHOD', kind='normal', doc='Readback reference method') 
+    cmd_ref_x_direct = Cpt(EpicsSignal, ':CMD:REF_X_DIRECT', kind='normal', doc='Set reference x-direction') 
+    ref_x_direct = Cpt(EpicsSignal, ':REF_X_DIRECT', kind='normal', doc='Readback reference x-direction') 
+    cmd_ref_y_direct = Cpt(EpicsSignal, ':CMD:REF_Y_DIRECT', kind='normal', doc='Set reference y-direction') 
+    ref_y_direct = Cpt(EpicsSignal, ':REF_Y_DIRECT', kind='normal', doc='Readback reference y-direction') 
+    cmd_ref_z_direct = Cpt(EpicsSignal, ':CMD:REF_Z_DIRECT', kind='normal', doc='Set reference z-direction') 
+    ref_z_direct = Cpt(EpicsSignal, ':REF_Z_DIRECT', kind='normal', doc='Readback reference z-direction')
+
+
+
+    """
+    Positions - Move
+    """
+    cmd_move = Cpt(EpicsSignal, ':CMD:MOVE', kind='normal')
+    cmd_x = Cpt(EpicsSignal, ':CMD:X', kind='normal')
+    cmd_y = Cpt(EpicsSignal, ':CMD:Y', kind='normal')
+    cmd_z = Cpt(EpicsSignal, ':CMD:Z', kind='normal')
+    cmd_rx = Cpt(EpicsSignal, ':CMD:RX', kind='normal')
+    cmd_ry = Cpt(EpicsSignal, ':CMD:RY', kind='normal')
+    cmd_rz = Cpt(EpicsSignal, ':CMD:RZ', kind='normal')
+
+
+
+    """
+    Stop
+    """
+    cmd_stop = Cpt(EpicsSignal, ':CMD:STOP', kind='normal')
+
+
+
+    """
+    Pose reachable?
+    """
+    cmd_reachable = Cpt(EpicsSignal, ':CMD:REACHABLE', kind='normal')
```

Comment:
These need some documentation strings attached to the component

Reply:
doc was added everywhere.